lib/internal/modules/esm/loader.js

```diff
@@ -109,6 +110,28 @@ let hooksProxy;
  * @typedef {ArrayBuffer|TypedArray|string} ModuleSource
  */
 
+/**
+ * @typedef {Object} LoadResult
+ * @property {ModuleFormat} format
+ * @property {ModuleSource} source
+ */
+
+/**
+ * @typedef {Object} ResolveResult
+ * @property {string} format
+ * @property {URL['href']} url
+ */
+
+/**
+ * @typedef {Object} Customizations
+ * @property {boolean} allowImportMetaResolve
+ * @property {(url: string, context: object) => Promise<LoadResult>} load
+ * @property {(originalSpecifier: string, parentURL: string, importAttributes: Record<string, string>) => Promise<ResolveResult>} resolve
+ * @property {(originalSpecifier: string, parentURL: string, importAttributes: Record<string, string>) => ResolveResult} resolveSync
+ * @property {(specifier: string, parentURL: string) => any} register
```

Comment:
Not sure if there's any rule as I haven't seem this used anywhere else, but for my TS syntax is more compact and easier to follow, and it is well supported in JSDoc

Reply:
This is what we do :)